broadlink/cover.py

```diff
@@ -58,3 +59,55 @@ def set_percentage_and_wait(self, new_percentage: int) -> None:
                 time.sleep(0.2)
                 current = self.get_percentage()
         self.stop()
+
+class dooya_new(device):
+    """Controls a Dooya curtain motor."""
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Initialize the controller."""
+        device.__init__(self, *args, **kwargs)
+        self.type = "Dooya DT360E(4F6E)"
+
+    def _send(self, magic1: int, magic2: int, magic3:int, magic4:int) -> int:
+        """Send a packet to the device."""
+        packet = bytearray(32)
+        packet[0] = 0x16
+        packet[2] = 0xa5
+        packet[3] = 0xa5
+        packet[4] = 0x5a
+        packet[5] = 0x5a
+        packet[6] = magic1
+        packet[7] = magic2
+        packet[8] = 0x02
+        packet[9] = 0x0b
+        packet[10] = 0x0a
+        packet[15] = magic3
+        packet[16] = magic4
+        response = self.send_packet(0x6a, packet)
+        check_error(response[0x22:0x24])
+        payload = self.decrypt(response[0x38:])
+        return payload[17]
+
+    def open(self) -> None:
+        """Open the curtain."""
+        self._send(0xc5, 0xc0, 0x01, 0x00)
+
+    def close(self) -> None:
+        """Close the curtain."""
+        self._send(0xc6, 0xc0, 0x02, 0x00)
+
+    def stop(self) -> None:
+        """Stop the curtain."""
+        self._send(0xc7, 0xc0, 0x03, 0x00)
+
+    def get_percentage(self) -> int:
+        """Return the position of the curtain."""
+        return self._send(0xca, 0xc0 ,0x06, 0x00)
+
+    def set_percentage_and_wait(self, new_percentage) -> None:
```

Comment:
I think this method should be renamed to `set_percentage` because it is non-blocking.

Reply:
`        return self._send(0xca, 0xc0 ,0x06, 0x00)

    def set_percentage(self, new_percentage) -> None:
        new_percent_hex = struct.pack('<H', 49357+new_percentage)`

Reply:
Does the device wait until the position is reached to send the response?